src/inference/src/dev/make_tensor.cpp

```diff
@@ -297,7 +297,27 @@ class RoiTensor : public ITensor {
     }
 
     void set_shape(ov::Shape new_shape) override {
-        OPENVINO_THROW("Shapes cannot be changed for ROI Tensor");
+        const auto tensor_rank = m_shape.size();
+        OPENVINO_ASSERT(new_shape.size() == tensor_rank);
+
+        auto owner_dim = m_owner->get_shape().cbegin();
+        auto stride = get_strides().cbegin();
+        auto new_dim = new_shape.cbegin();
+        auto dim = m_shape.cbegin();
+
+        for (size_t current_offset = m_offset; new_dim != new_shape.cend(); ++dim, ++new_dim, ++owner_dim, ++stride) {
+            const auto roi_coord = (current_offset / *stride);
```

Comment:
do we need to take `m_offset` into account? 
maybe we can store initial `m_shape`  as `m_capacity` as well? and will check elementwise that `m_capacity` greater equal than `new_shape` ?

Reply:
>  with current solution we can resize to the tensor shape.

Do you mean we can `set_shape` to be out of initial ROI? then it's prohibited.

Reply:
Yes, ok so clear now and will apply changes.